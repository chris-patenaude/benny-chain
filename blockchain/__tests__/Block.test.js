const { DIFFICULTY } = require("../../config");
const Block = require("../block");
const NOW = 123456789;
const CREATION = 1641801600000;

describe("The Block class", () => {
    let data, genesisBlock, block;
    beforeEach(() => {
        jest.spyOn(global.Date, "now").mockImplementation(() => NOW);
        data = [1, 2, 3];
        genesisBlock = Block.genesis();
    });
    it("Should print out a text description of an individual instance", () => {
        expect(genesisBlock.toString()).toMatchSnapshot();
    });
    it("Should generate a genesis block for initializing the blockchain", () => {
        expect(genesisBlock.data).toHaveLength(0);
        expect(genesisBlock.hash).toEqual("genesis-20221001");
        expect(genesisBlock.lastHash).toEqual("---");
        expect(genesisBlock.timestamp).toEqual(CREATION);
    });
    it("Should generate a new block with a valid hash and nonce value", () => {
        block = Block.mineBlock(genesisBlock, data);
        expect(block.data).toEqual(data);
        expect(block.hash).toMatchSnapshot();
        expect(block.lastHash).toEqual(genesisBlock.hash);
        expect(block.timestamp).toEqual(NOW);
        expect(block.hash.substring(0, DIFFICULTY)).toEqual("0".repeat(DIFFICULTY));
    });
});
