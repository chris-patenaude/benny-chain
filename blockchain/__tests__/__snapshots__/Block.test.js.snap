// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Block class Should generate a new block with a valid hash and nonce value 1`] = `"000644a58c6dd323b2e9a6c652fe43ac21d5f82c51c28939dbe45f474098efa5"`;

exports[`The Block class Should print out a text description of an individual instance 1`] = `
"Block - 
		Timestamp: 1641801600000
		Last Hash: ---...
		Hash     : genesis-20...
		Nonce    : 0
		Data     : "
`;
